src/main/java/com/fasterxml/jackson/dataformat/xml/util/TypeUtil.java

```diff
@@ -12,8 +14,11 @@
      */
     public static boolean isIndexedType(JavaType type)
     {
-        if (type.isContainerType()) {
-            Class<?> cls = type.getRawClass();
+        Class<?> cls = type.getRawClass();
+        if (type.isContainerType()
+            || Iterator.class.isAssignableFrom(cls)
+            || Stream.class.isAssignableFrom(cls)
```

Comment:
is this needed? I'd have expected Iterator and Stream to have JavaTypes that are container types.

Reply:
@cowtowncoder would `isCollectionLikeType()` something that could be considered here? `isContainerType()` does not include Streams and Iterators. Since we have a `JavaType` instance here, it would be nice not to have to hardcode the classes.

Reply:
Jackson supports other languages like Scala and Kotlin. The JavaType abstraction is a much more extensible way to do this logic than hardcoding random Java classes.

Reply:
We may have no choice but to do the code as you are doing it. It would be interesting to get @cowtowncoder's input.

Reply:
I am trying to figure out how I feel about this. I feel Iterators etc are not "Collection"-like; perhaps we could/should consider new `IteratorType` for 2.16? It would have element type; not sure what other commonality there should be.

Trick here (well one of them) is to figure out how Scala, Kotlin (and maybe later JDK modules) would augment information -- otherwise we could just add something in `SimpleType`.

For inspiration we could look into `ReferenceType` (or `CollectionLikeType`).

There could then be something like `isIterableType()` in `JavaType` returning true for Map(Like), Collection(Like), Array and IteratorTypes.

Would this be a way forward?

Reply:
Maybe we could merge this PR as it is and then proceed with the JavaType changes independently? We can always uptake the JavaType changes for this use case later.

Reply:
I see that @cowtowncoder made this point about the JavaType changes in another comment.

Reply:
Right, +1 for separate approach.

As to `IteratorType`, could also consider `IterableType` (although it has same possible issue wrt naming).
Perhaps `IterationType` would be a better choice.

Also on `MapLikeType`, `CollectionLikeType` -- these would not be super types; Maps are not iterators (nor Collections; nor MapLike/CollectionLike). Partly since not all Map(like) types implement actual iterator/iterable (in fact, most don't).
Or put another way: in case something implements `Iterator` (etc) but is also `Map` (or Map-like), our `JavaType` would be `Map[Like]Type`. Iterator tends to be more an add-on (or mix-in) type.